### PR TITLE
Task notification endpoint 

### PR DESCRIPTION
**Pull request by bbrands02**

### CODE REVIEW

File: src/Service/NotificationToTaakService.php [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+/**
+ * This class handles the synchronization of a notification of a zaaksysteem task taken to a customerinteractionbundle taak.
+ *
+ * Fetches all tasks of the case of the notification.
+ *
+ * @author  Conduction BV <info@conduction.nl>, Barry Brands <barry@conduction.nl>
+ * @license EUPL-1.2 https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * @category Service
+ */
```
Comment (WilcoLouwerse) [inline src/Service/NotificationToTaakService.php:11]: docs should probably be places above the 'class' line (25)

File: src/Service/NotificationToTaakService.php [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+use Psr\Log\LoggerInterface;
+use Ramsay\Uuid;
+
+class NotificationToTaakService
+{
+
+    /**
+     * @var array
+     */
+    private array $configuration;
```
Comment (WilcoLouwerse) [inline src/Service/NotificationToTaakService.php:31]: We don't want these anymore 😝 just pass parameters in your functions

File: src/Service/NotificationToTaakService.php [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        // Get taak schema.
+        $schema = $this->resourceService->getSchema(reference: $this->configuration['schema'], pluginName: $pluginName);
+        if ($schema === null) {
+            return $this->data;
+        }
+
+        // Get task to taak mapping.
+        $mapping = $this->resourceService->getMapping(reference: $this->configuration['mapping'], pluginName: $pluginName);
+        if ($mapping === null) {
+            return $this->data;
```
Comment (WilcoLouwerse) [inline src/Service/NotificationToTaakService.php:82]: optional: you could combine these 3 if statements above^

File: src/Service/NotificationToTaakService.php [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+        $endpoint = $this->configuration['endpoint']."?filter[relationships.case.id]=".$this->data['case_uuid'];
+
+        // Fetch all tasks of the case
+        try {
+            $this->pluginLogger->info("Fetching tasks for case id: {$this->data['case_uuid']}..");
+            $response = $this->callService->call($source, $endpoint, 'GET', [], false, false);
+            $tasks    = $this->callService->decodeResponse(source: $source, response: $response);
+        } catch (Exception $e) {
+            // isset($this->style) === true && $this->style->error("Failed to fetch case: $caseID, message:  {$e->getMessage()}");
```
Comment (WilcoLouwerse) [inline src/Service/NotificationToTaakService.php:93]: remove this :)

File: src/Service/NotificationToTaakService.php [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+        if (isset($taskWeNeed) === false) {
+            $this->pluginLogger->error("Could not find the correct task ({$this->data['entity_id']}) in the tasks of the case ({$this->data['case_uuid']})");
+
+            return $this->data;
+        }
+
+        // Synchronize correct task.
+        $synchronization = $this->oldSynchronizationService->findSyncBySource(source: $source, entity: $schema, sourceId: $taskWeNeed['id'], endpoint: $endpoint);
+        $synchronization = $this->oldSynchronizationService->synchronize(synchronization: $synchronization, sourceObject: $taskWeNeed, unsafe: false, mapping: $mapping);
```
Comment (WilcoLouwerse) [inline src/Service/NotificationToTaakService.php:114]: Optional: Maybe add some new lines to make this more readable

**Review by WilcoLouwerse — CHANGES_REQUESTED** [review]

File: Installation/Schema/xxllnctoktb.notification.schema.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{
+    "$id": "https://commongateway.nl/schema/xxllnctoktb.notification.schema.json",
+    "$schema": "https://docs.commongateway.nl/schemas/Entity.schema.json",
+    "version": "0.1",
+    "title": "Notification",
+    "description": "A notification object from xxllnc",
+    "type": "object",
+    "properties": {
+        "changes": {
```
[the remark this answers is not in the document]
Comment (bbrands02) [reply Installation/Schema/xxllnctoktb.notification.schema.json:9]: see new added properties

**Review by WilcoLouwerse — APPROVED** [review]
Pls dubbel check if you are using correct SyncService uses

File: src/Service/NotificationToTaakService.php [hunk 7]
```diff
@@ -2,9 +2,9 @@
 
 namespace CommonGateway\XxllncToKTBBundle\Service;
 
-use CommonGateway\CoreBundle\Service\SynchronizationService;
+use App\Service\SynchronizationService;
```
[the remark this answers is not in the document]
Comment (bbrands02) [reply src/Service/NotificationToTaakService.php:5]: No CoreBundle only has synchronizeTemp function for outgoing syncs

**Review by WilcoLouwerse — APPROVED** [review]

**Pull request merged**